src/PasswordPolicy.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+namespace Drupal\password_enhancements;
+
+use Drupal\password_enhancements\Exception\PolicyInvalidArgumentException;
+use Drupal\user\Entity\Role;
+use Drupal\user\RoleInterface;
+
+/**
+ * Defines the Password Policy.
+ */
+final class PasswordPolicy implements PasswordPolicyInterface {
+
+  /**
+   * Expiry of the password in seconds, 0 means that passwords doesn't expire.
+   *
+   * @var int
+   */
+  private $expireSeconds;
+
+  /**
+   * Show warning message before the password would expire in a given seconds.
+   *
+   * @var int
+   */
+  private $expireWarnSeconds;
+
+  /**
+   * Expiry warning message.
+   *
+   * @var string|null
```

Comment:
Because constructor's parameter enforces `string $expiry_warning_message` this cannot be null.

Reply:
As the expiry warning message CAN be `NULL`, I'm fixing the constructor's parameter instead.